src/triton_cli/parser.py

```diff
@@ -496,8 +465,97 @@ def parse_args(argv=None):
     parse_args_repo(subcommands)
     parse_args_server(subcommands)
     parse_args_inference(subcommands)
+    parse_args_optimize(subcommands)
     parse_args_profile(subcommands)
     parse_args_utils(subcommands)
     add_verbose_args([parser])
-    args = parser.parse_args(argv)
+
+    argv_ = argv if argv is not None else sys.argv[1:]
+    # If a passthrough command is passed as the first arg,
+    # special handling is needed.
+    need_special_handling = len(argv_) > 1 and argv_[0] in PASSTHROUGH_SUBCOMMANDS
+    if need_special_handling:
+        updated_args, pruned_args = prune_extra_subcommand_args(
+            argv_, subcommands.choices.keys()
+        )
+        args, unknown_args = parser.parse_known_args(updated_args)
+        args = add_unknown_args_to_args(args, pruned_args, unknown_args)
+    else:
+        args = parser.parse_args(argv)
     return args
+
+
+# ================================================
+# Helper functions
+# ================================================
+def build_command(executable: str, args: argparse.Namespace):
+    skip_args = ["func", "task"]
+    cmd = [executable]
+    program_subcommand = None
+    for arg, value in vars(args).items():
+        if arg in skip_args:
+            pass
+        elif arg in PASSTHROUGH_SUBCOMMANDS:
+            program_subcommand = arg
+        elif value is False:
+            pass
+        elif value is True:
+            if len(arg) == 1:
+                cmd += [f"-{arg}"]
+            else:
+                cmd += [f"--{arg}"]
+        else:
+            if len(arg) == 1:
+                cmd += [f"-{arg}", f"{value}"]
+            else:
+                arg = arg
+                cmd += [f"--{arg}", f"{value}"]
```

Comment:
```suggestion
            cmd = cmd + [f"-{arg}", f"{value}"] if len(arg)==1 else cmd + [f"--{arg}", f"{value}"]
```
Maybe same thing here? If not, I think we can get rid of the `arg=arg` statement.

Reply:
That's an embarrassing oversight. Thanks for catching this! Removed.

Reply:
Left a comment above regarding the conditional style.

Reply:
Not embarrassing; that's why we have reviews! :)